modules/hdf5_handler/h5cfdaputil.cc

```diff
@@ -328,35 +328,43 @@ HDF5CFDAPUtil:: print_attr(H5DataType type, int loc, void *vals)
     case H5FLOAT32:
         {
             float attr_val = *(float*)vals;
-            bool is_a_fin = isfinite(attr_val);
-            gp.fp = (float *) vals;
-            rep << showpoint;
-            rep << setprecision(10);
-            rep << *(gp.fp+loc);
-            string tmp_rep_str = rep.str();
-            if (tmp_rep_str.find('.') == string::npos
-                && tmp_rep_str.find('e') == string::npos
-                && tmp_rep_str.find('E') == string::npos
-                && (true == is_a_fin)){
-                rep<<".";
+            if (isnan(attr_val)) 
+                rep<<"NaN";
+            else {
+                bool is_a_fin = isfinite(attr_val);
+                gp.fp = (float *) vals;
+                rep << showpoint;
+                rep << setprecision(10);
+                rep << *(gp.fp+loc);
+                string tmp_rep_str = rep.str();
```

Comment:
I'm not sure about the rep << showpoint, ..., because I need more context, but the repeated calls to find() can be collapsed into 'rep.str().find_first_of(".eE") == string::npos' and I think the 'isfinite(attr_val)' can be moved dow into the if, so that it can be rewritten as:

if ((rep.str().find_first_of(".eE") == string::npos) && isfinite(attr_val)) {
    rep << ".";
}

It might not make that big a difference, but it might be something that would ease future work if you need to do this in the future.

Reply:
All these are originally codes. The showpoint and the setprecision are necessary but I cannot remember why. I will trim the code with your suggestions. Thanks,